From 80519bf4d08250e6cdaa0deba518d5a9ffe4c184 Mon Sep 17 00:00:00 2001
From: Simon McVittie <simon.mcvittie@collabora.co.uk>
Date: Mon, 27 Jul 2009 21:21:15 +0100
Subject: [PATCH 3/3] Don't advertise support for credentials-passing on platforms where it won't work

---
 src/tube-stream.c |   11 +++++++++--
 1 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/src/tube-stream.c b/src/tube-stream.c
index e71fe82..e39532f 100644
--- a/src/tube-stream.c
+++ b/src/tube-stream.c
@@ -2358,8 +2358,15 @@ gabble_tube_stream_get_supported_socket_types (void)
       1);
   access_control = TP_SOCKET_ACCESS_CONTROL_LOCALHOST;
   g_array_append_val (unix_tab, access_control);
-  access_control = TP_SOCKET_ACCESS_CONTROL_CREDENTIALS;
-  g_array_append_val (unix_tab, access_control);
+
+  /* Credentials-passing is non-portable, so only advertise it on platforms
+   * where we have an implementation (like Linux) */
+  if (gibber_unix_transport_supports_credentials ())
+    {
+      access_control = TP_SOCKET_ACCESS_CONTROL_CREDENTIALS;
+      g_array_append_val (unix_tab, access_control);
+    }
+
   g_hash_table_insert (ret, GUINT_TO_POINTER (TP_SOCKET_ADDRESS_TYPE_UNIX),
       unix_tab);
 
