From bced2f61312ab407964ac3ef0653183eb3b8c6cc Mon Sep 17 00:00:00 2001
From: Marcus Lundblad <ml@update.uu.se>
Date: Tue, 15 May 2012 22:40:04 +0200
Subject: [PATCH 1/2] Make Android 4.0 GTalk contacts appear as capable of a/v
 calls.

Reviewed-by: Sjoerd Simons <sjoerd@greynoise.nl>
Bug: https://bugs.freedesktop.org/show_bug.cgi?id=36998
---
 src/presence-cache.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/presence-cache.c b/src/presence-cache.c
index 22e025c..b5a0dab 100644
--- a/src/presence-cache.c
+++ b/src/presence-cache.c
@@ -509,6 +509,8 @@ gabble_presence_cache_add_bundles (GabblePresenceCache *cache)
   gabble_presence_cache_add_bundle_caps (cache, \
       "http://www.android.com/gtalk/client/caps#" cap, features); \
   gabble_presence_cache_add_bundle_caps (cache, \
+      "http://www.android.com/gtalk/client/caps2#" cap, features); \
+  gabble_presence_cache_add_bundle_caps (cache, \
       "http://talk.google.com/xmpp/bot/caps#" cap, features);
 
   /* Cache various bundle from the Google Talk clients as trusted.  Some old
-- 
1.7.10

